ios/FluentUI/Command Bar/CommandBar.swift

```diff
@@ -0,0 +1,287 @@
+    }()
+
+    private lazy var stackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: buttonGroups)
+
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        stackView.spacing = Constants.buttonSpacing
+
+        return stackView
+    }()
+
+    private lazy var buttonGroups: [CommandBarButtonGroupView] = {
+        itemGroups.map { items in
+            CommandBarButtonGroupView(buttons: items.compactMap { item in
+                guard let button = itemsToButtonsMap[item] else {
+                    fatalError("Button is not initialized in commandsToButtons")
+                }
+
+                return button
+            })
+        }
+    }()
+
+    private var leadingButton: CommandBarButton?
+    private var trailingButton: CommandBarButton?
+
+    private let containerMaskLayer: CAGradientLayer = {
+        let layer = CAGradientLayer()
+        layer.colors = [UIColor.clear, UIColor.white, UIColor.white, UIColor.clear].map { $0.cgColor }
+        layer.startPoint = CGPoint(x: 0, y: 0.5)
+        layer.endPoint = CGPoint(x: 1, y: 0.5)
+        layer.locations = [0, 0, 1]
+
+        return layer
+    }()
+
+    // MARK: - Init
+
+    public init(itemGroups: [CommandBarItemGroup], leadingItem: CommandBarItem? = nil, trailingItem: CommandBarItem? = nil) {
+        self.itemGroups = itemGroups
+
+        super.init(frame: .zero)
+
+        if let leadingItem = leadingItem {
+            self.leadingButton = button(forItem: leadingItem, isFixed: true)
+        }
+        if let trailingItem = trailingItem {
+            self.trailingButton = button(forItem: trailingItem, isFixed: true)
+        }
+
+        translatesAutoresizingMaskIntoConstraints = false
+
+        configureHierarchy()
+        updateButtonsState()
+    }
+
+    @available(*, unavailable)
+    public required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public override var intrinsicContentSize: CGSize {
+        .zero
+    }
+
+    // MARK: Overrides
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+
+        containerMaskLayer.frame = containerView.bounds
+        updateContainerMask()
+    }
+
+    // MARK: - Public methods
+
+    @objc public func updateButtonsState() {
+        for button in itemsToButtonsMap.values {
+            button.updateState()
+        }
+    }
+}
+
+// MARK: - Scroll view delegate
+
+extension CommandBar: UIScrollViewDelegate {
+    public func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        updateContainerMask()
+    }
+}
+
+// MARK: - Private methods
+
+private extension CommandBar {
+    struct Constants {
+        static let fadeViewWidth: CGFloat = 24
+        static let horizontalButtonInset: CGFloat = 4
+        static let buttonSpacing: CGFloat = 16.0
+        static let insets = UIEdgeInsets(top: 8.0, left: 8.0, bottom: 8.0, right: 8.0)
+    }
```

Comment:
I know we have a bunch of this in our code, but we're moving away from structs that are just constants in favor of just having constants at the bottom of our files